app/src/main/java/com/socketmobile/stockCount/ui/OptionsActivity.kt

```diff
@@ -31,6 +53,48 @@ class OptionsActivity : AppCompatActivity() {
             val i = Intent(Intent.ACTION_VIEW, Uri.parse(getString(R.string.sdk_github_url)))
             startActivity(i)
         }
+        buyScannerButton.setOnClickListener {
+            val progressDialog = ProgressDialog(this)
+            progressDialog.show()
+            val country = Locale.getDefault().country
+
+            if (country == "CN") {
+                openLink(ChinaLink)
+                return@setOnClickListener
+            }
+
+            val keyObj = getRefersionKey(Locale.getDefault().country)
+            val param = hashMapOf(
+                    "refersion_public_key" to keyObj.pub_key,
+                    "refersion_secret_key" to keyObj.sec_key,
+                    "affiliate_code" to keyObj.aff_code
+            )
+            val call = refersionService.getAffiliate(param)
+            call.enqueue(object: Callback<Affiliate> {
+                override fun onFailure(call: Call<Affiliate>, t: Throwable) {
+                    progressDialog.hide()
+                    AlertDialog.Builder(this@OptionsActivity)
+                            .setMessage(t.message)
```

Comment:
should this message get localized?

Reply:
This message comes from SDK or android system. not from our app.
I think no way to translate it on our app.